lib/routes/client/register.js

```diff
@@ -50,16 +50,34 @@ module.exports = {
       canGrant: !!payload.can_grant,
       whitelisted: !!payload.whitelisted
     };
-    db.registerClient(client).then(function() {
-      reply({
-        id: hex(client.id),
-        secret: hex(secret),
-        name: client.name,
-        redirect_uri: client.redirectUri,
-        image_uri: client.imageUri,
-        can_grant: client.canGrant,
-        whitelisted: client.whitelisted
-      }).code(201);
-    }, reply);
+    var developerEmail = req.auth.credentials.email;
+    var developerId = null;
+
+    return db.getDeveloper(developerEmail)
+      .then(function (developer) {
+
+        // must be a developer to register clients
+        if (! developer) {
+          return reply(AppError.unauthorized('Illegal Developer'));
```

Comment:
I'd change this to `throw AppError.unauthorized`. This will make it skip all the other `then` paths, and just use the last error path, which at L81 will `reply` with the error. Otherwise, since this returns the promise with a non error, it will execute the rest of the `then`s.